node/src/chain/header_validation.rs

```diff
@@ -191,34 +193,35 @@ impl<'a, DB: database::DB> Validator<'a, DB> {
 }
 
 pub async fn verify_block_cert(
+    prev_block_hash: [u8; 32],
     curr_seed: Signature,
     curr_eligible_provisioners: &Provisioners,
     block_hash: [u8; 32],
-    height: u64,
+    round: u64,
     cert: &ledger::Certificate,
     iteration: u8,
-    enable_quorum_check: bool,
 ) -> anyhow::Result<(QuorumResult, QuorumResult)> {
     let committee = RwLock::new(CommitteeSet::new(curr_eligible_provisioners));
-
-    let hdr = node_data::message::Header {
-        topic: node_data::message::Topics::Unknown,
-        pubkey_bls: node_data::bls::PublicKey::default(),
-        round: height,
-        iteration,
-        block_hash,
+    let vote = match block_hash == [0; 32] {
```

Comment:
For the next PR `vote` should be passed as argument or even included in the certificate.